atom/browser/ui/win/taskbar_host.cc

```diff
@@ -117,17 +117,39 @@ bool TaskbarHost::SetThumbarButtons(
   return SUCCEEDED(r);
 }
 
-bool TaskbarHost::SetProgressBar(HWND window, double value) {
+bool TaskbarHost::SetProgressBar(
+  HWND window, double value, const std::string& mode) {
```

Comment:
I think we should map the string value to an enum in `atom_api_window.cc` and pass the enum here.